Source/WebCore/platform/mediastream/mac/CoreAudioSharedUnit.cpp

```diff
@@ -681,8 +682,30 @@ bool CoreAudioSharedUnit::migrateToNewDefaultDevice(const CaptureDevice& capture
     handleNewCurrentMicrophoneDevice(WTFMove(*device));
     return true;
 }
-#endif
 
+void CoreAudioSharedUnit::prewarmAudioUnitCreation(CompletionHandler<void()>&& callback)
+{
+    if (!m_audioUnitCreationWarmupPromise) {
+        m_audioUnitCreationWarmupPromise = invokeAsync(WorkQueue::create("CoreAudioSharedUnit AudioUnit creation").get(), [weakThis = WeakPtr { *this }] () mutable {
+            callOnMainRunLoop([weakThis = WTFMove(weakThis), vpioUnitOrError = createAudioUnit(true)] () mutable {
```

Comment:
I would still ditch `callOnMainRunLoop` here and use `whenSettled` on the promise returned by invokeAsync. with #20770 you can just do `return createAudioUnit(true)` in the lambda.
```
m_audioUnitCreationWarmupPromise = invokeAsync(WorkQueue::create("CoreAudioSharedUnit AudioUnit creation").get(), [] {
        return createAudioUnit(true);
    })->whenSettled(RunLoop::main(), [weakThis = WeakPtr { *this }](auto&& vpioUnitOrError) mutable {
        if (!weakThis)
            return GenericNonExclusivePromise::createAndReject();
        if (vpioUnitOrError.has_value())
            weakThis->setStoredVPIOUnit(WTFMove(vpioUnitOrError.value()));
        return GenericNonExclusivePromise::createAndResolve();
    });
```

Reply:
I should add that to me `m_audioUnitCreationWarmupPromise` is the promise that the AU has been created *and* is ready for use (that is got stored).

By dissociating the call to `setStoredVPIOUnit` having been run from the AU creation promise, it becomes much less clear that the AU is not just created but also usable by the CoreAudioSharedUnit users.

It is an implementation detail that the end result is the same in how the task are dispatched.